proposals/0089-programify-feature-gate-program.md

```diff
@@ -0,0 +1,149 @@
+is irreversible; if the runtime knows about a feature gate at some address, it
+will activate it at the next epoch boundary. This means there is no recourse in
+the case of a mistaken queuing, discovery of a bug, or simply a desire to manage
+the cadence and schedule of activations.
+
+A fully-implemented Core BPF program would take ownership of those accounts and
+support revoking queued features, giving engineers more flexibility and
+safeguards.
+
+## Alternatives Considered
+
+The Feature Gate program could instead be implemented as a built-in native
+program, rather than a Core BPF program. However, this would mean any changes to
+the program would need to be implemented by all validator clients in
+coordination. This makes upgrading the program cumbersome and potentially
+dangerous.
+
+With the Feature Gate program instead implemented as a Core BPF program, the
+program could be changed only once, eliminating this duplication of work.
+
+## New Terminology
+
+- Feature Gate program: The Core BPF program that all feature accounts will be
+  assigned to, with address `Feature111111111111111111111111111111111111`.
+- "Revoke" or "revoke pending activation": The act of reallocating a feature
+  account's data to zero, assigning it to the System Program, and defunding its
+  lamports balance - effectively removing it from the runtime's recognized set
+  of pending feature activations.
+
+## Detailed Design
+
+The Feature Gate program shall be a Core BPF program whose upgrade authority
+will be controlled by a multi-sig authority, with keyholders from each validator
+client implementation. This includes Solana Labs, Jito, and Jump. In the future,
+this can be expanded to include new clients such as Syndica.
+
+The program shall initially be designed to support one instruction:
+`RevokePendingActivation`. Any other instructions or functionality this program
+may support in the future is outside the scope of this SIMD.
+
+As mentioned above under "New Terminology", when this instruction is invoked by
+a feature key-holder, the program will reallocate the account to zero, assign it
+back to the System Program, and defund its lamports balance. As a result, the
+runtime will no longer recognize this feature as pending, since it will no
+longer be owned by `Feature111111111111111111111111111111111111`.
+
+Consider the instruction as it may appear in the Feature Gate program:
+
+```rust
+pub enum FeatureGateInstruction {
+    /// Revoke a pending feature activation.
+    ///
+    /// A "pending" feature activation is a feature account that has been
+    /// allocated and assigned, but hasn't yet been updated by the runtime
+    /// with an `activation_slot`.
+    ///
+    /// Features that _have_ been activated by the runtime cannot be revoked.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w+s]`    Feature account
+    ///   1. `[w]`      Destination (for rent lamports)
+    RevokePendingActivation,
+}
+```
```

Comment:
I don't really see much point in this being here. Since we generally try to avoid code in SIMDs, I think just remove it.

Reply:
Sorry, I thought you guys specifically said to include state and instruction layouts in the SIMDs.

Reply:
Circling back here, I plucked the code snippet as requested!